runbot_gitlab/models/runbot_repo.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+# Copyright 2010 - 2014 Savoir-faire Linux (<http://www.savoirfairelinux.com>).
+# Copyright 2017 Vauxoo <info@vauxoo.com> (<https://www.vauxoo.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import os
+import logging
+import re
+import urllib
+
+import requests
+
+from openerp import models, fields
+
+_logger = logging.getLogger(__name__)
+
+
+def _get_url(url, base):
+    match_object = re.search('([^/]+)/([^/]+)/([^/.]+(.git)?)', base)
```

Comment:
I'm honestly lost on this method. Would it be possible to maybe use named groups? I think that might help the readability.